config/initializers/errors.rb

```diff
@@ -12,5 +12,37 @@ def initialize(message="Invalid SQL query for this Live Query")
         super
       end
     end
+
+    class ExpectedError < StandardError
+      attr_reader :error_code, :status, :message
+
+      def initialize(error_code, status, message)
+        @error_code = error_code
+        @status = status
+        @message = message
+      end
+
+      def display_error()
+        recursively_print(self)
+      end
+
+      def recursively_print(error, margin="")
+        FOREST_LOGGER.info "#{margin}#{error.message}"
+
+        # NOTICE: Ruby < 2.1.0 doesn't have `cause`
+        if error.respond_to?(:cause) && !error.cause.nil?
+          FOREST_LOGGER.info "#{margin}which was caused by:"
```

Comment:
Why not `FOREST_LOGGER.error`?

Reply:
Because if a user us not authorized, it is not an error in the perspective of the liana server. This is like the expected/unexpected errors in forestadmin-server.